packages/engine/Source/Shaders/Voxels/VoxelUtils.glsl

```diff
@@ -0,0 +1,27 @@
+struct Ray {
+    vec3 pos;
+    vec3 dir;
+    vec3 rawDir;
+};
+
+#if defined(JITTER)
+float hash(vec2 p)
+{
+    vec3 p3 = fract(vec3(p.xyx) * 50.0); // magic number = hashscale
```

Comment:
I know this was existing code that was just moved to this file in this PR, but would it be possible to better explain what these magic numbers are in a comment?

Reply:
I haven't been able to trace where this hash function came from. It is similar to one of the functions in a [popular shadertoy example](https://www.shadertoy.com/view/4djSRW), but most uses that I have seen use a different value for the hashscale. See for example the `hash12` function in [this gist](https://gist.github.com/lhog/cfc256a0486d35e7dd90bc4970e1ebc0).
I added a link to the shadertoy in a comment.